app/services/katello/pulp/yum_metadata_file.rb

```diff
@@ -7,8 +7,10 @@ def update_model(model)
         shared_attributes = backend_data.keys & model.class.column_names
         shared_json = backend_data.select { |key, _v| shared_attributes.include?(key) }
         repo = ::Katello::Repository.find_by(:pulp_id => backend_data['repo_id']).try(:id)
-        model.update!(shared_json.merge(repository_id: repo,
-                                                  name: find_name_from_json(backend_data)))
+        filename = find_name_from_json(backend_data)
+        model.update!(shared_json.merge(repository_id: repo, name: filename))
+        Katello::YumMetadataFile.where("id != ? AND repository_id = ? AND name = ? AND checksum = ?",
+                                       model.id, repo, filename, model.checksum).delete_all if repo
```

Comment:
looks like there's a rubocop error:
```
[2021-05-07T06:38:12.432Z] /home/jenkins/workspace/katello-pr-test/app/services/katello/pulp/yum_metadata_file.rb:12:9: C: Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement.
[2021-05-07T06:38:12.432Z]         Katello::YumMetadataFile.where("id != ? AND repository_id = ? AND name = ? AND checksum = ?", ...
[2021-05-07T06:38:12.432Z]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It wants it wrapped like this:

if repo
        Katello::YumMetadataFile.where("id != ? AND repository_id = ? AND name = ? AND checksum = ?",
                                       model.id, repo, filename, model.checksum)
end

Reply:
Can you explain why it deleting these yum metadata files? Is the purpose to delete duplicates?

Reply:
Yes. When publishing a content view, "index_content" will copy the yum metadata files in katello db from the source repo to the destination repo (skipping the import from pulp). It is not correct for katello to clone the yum metadata files this way. Unlike rpms and other unit types, Pulp creates a new yum metadata file unit per repo even the same file already existed in another repo. But i think it is not a big deal for Katello since it only cares the checksum and file type. This code is to delete the wrong yum metadata files and avoid duplicate when running reimport.

Example:
```
[#<Katello::YumMetadataFile:XXXXXXXXXXXXX
  id: XXXX,
  pulp_id: "25589690-8c70-4043-a10b-9a11cb6539b2",  <======== Unit copied from source repo during index_content. The code will delete this.
  checksum: "be129c6404fea58b6123a1e707805ca7736ae8c8",
  name: "productid",
  repository_id: 123456,
  ...
  migrated_pulp3_href: nil>,
 #<Katello::YumMetadataFile:XXXXXXXXXXXXX
  id: XXXX,
  pulp_id: "c9bc8f14-cdae-405f-bb56-37270d440b72",   <======== Unit that actually associated to the dest repo in Pulp. It is added when running reimport script.
  checksum: "be129c6404fea58b6123a1e707805ca7736ae8c8",
  name: "productid",
  repository_id: 123456,
  ...
  migrated_pulp3_href: nil>]
```

Reply:
Fixed rubocop suggestion